tests/test_app/TestApp/Application.php

```diff
@@ -74,6 +74,12 @@ public function middleware(MiddlewareQueue $middlewareQueue): MiddlewareQueue
      */
     public function routes(RouteBuilder $routes): void
     {
+        // Additional routes to load
+        if (Configure::check('TestApp.routes')) {
```

Comment:
@markstory  Using `TestCase::loadPlugins()` to update `TestCase::$appPluginsToLoad` didn't have any issues with duplicate plugin loading but instead had errors due to duplicate routes loading instead (trying to reconnect a route with the same name throws exception) :slightly_smiling_face:.

This is because `loadPlugins()` loads the plugin routes, so later when an integration test is run and app instance loads the plugins routes again it causes errors. I mitigated that by clearing the routes in `ContainerStubTrait::createApp()` but that caused failure in `IntegrationTestTraitTest` since it directly connects routes which got cleared.

My solution to the above problem was to add the `TestApp.routes` config, to allow setting a closure which connects the routes from the test app class.

Reply:
Hopefully there aren't many plugins that are creating routes with a builder too.